### PR TITLE
Add notes about invoking adapter methods in Jinja space

### PR DESCRIPTION
**Pull request by VersusFacit**
## Description & motivation
Followup to [this core PR](https://github.com/dbt-labs/dbt-core/pull/5306). The idea is to include more of explanation of what `adapter` does for adapter maintainers.

### CODE REVIEW

File: website/docs/reference/dbt-jinja-functions/adapter.md [hunk 1]
```diff
@@ -5,9 +5,13 @@ id: "adapter"
 
 ## Overview
 
-`adapter` is a wrapper around the internal database adapter used by dbt. It allows users to make calls to the database in their dbt models. The adapter methods below will be translated into specific SQL statements depending on the type of adapter your project is using.
+dbt implements database communication using an internal database adapter object (e.g. BaseAdapter, SnowflakeAdapter). The Jinja object `adapter` is a wrapper around this internal database adapter object.
```
Comment (runleonarun) [inline website/docs/reference/dbt-jinja-functions/adapter.md]:
```suggestion
dbt implements database communication using an internal database adapter object. For example, BaseAdapter and SnowflakeAdapter. The Jinja object `adapter` is a wrapper around this internal database adapter object.
```

File: website/docs/reference/dbt-jinja-functions/adapter.md [hunk 2]
```diff
@@ -5,9 +5,13 @@ id: "adapter"
 
 ## Overview
```
Comment (runleonarun) [inline website/docs/reference/dbt-jinja-functions/adapter.md]:
```suggestion
```

**Review by runleonarun — DISMISSED** [review]
This looks good! I will add my feedback and merge. Thanks for adding the context!

**Review by runleonarun — DISMISSED** [review]
🚢

File: website/docs/reference/dbt-jinja-functions/adapter.md [hunk 3]
```diff
@@ -3,11 +3,14 @@ title: "adapter"
 id: "adapter"
 ---
 
-## Overview
 
-`adapter` is a wrapper around the internal database adapter used by dbt. It allows users to make calls to the database in their dbt models. The adapter methods below will be translated into specific SQL statements depending on the type of adapter your project is using.
+dbt implements database communication using an internal database adapter object. For example, BaseAdapter and SnowflakeAdapter. The Jinja object `adapter` is a wrapper around this internal database adapter object.
```
Comment (runleonarun) [inline website/docs/reference/dbt-jinja-functions/adapter.md]:
```suggestion
Your database communicates with dbt using an internal database adapter object. For example, BaseAdapter and SnowflakeAdapter. The Jinja object `adapter` is a wrapper around this internal database adapter object.
```

**Comment by runleonarun** [comment]
@dataders How do these changes look? Can we merge this old PR?

**Review by dataders — APPROVED** [review]
so grateful for this addition @VersusFacit ! Anders from two years ago would have loved this information. I think it's good to go. for extra credit, you could find a good place in the adapter development docs to link to this page and your useful addition!

**Pull request merged**